ntp-proto/src/algorithm/kalman/peer.rs

```diff
@@ -0,0 +1,1067 @@
+/// This module implements a kalman filter to filter the measurements
+/// provided by the peers.
+///
+/// The filter tracks the time difference between the local and remote
+/// timescales. For ease of implementation, it actually is programmed
+/// mostly as if the local timescale is absolute truth, and the remote
+/// timescale is the one that is estimated. The filter state is kept at
+/// a local timestamp t, and progressed in time as needed for processing
+/// measurements and producing estimation outputs.
+///
+/// The state is a vector (D, w) where
+///  - d is the offset between the remote and local timescales
+///  - w is (in seconds per second) the frequency difference.
+///
+/// For process noise, we assume this is fully resultant from frequency
+/// drift between the local and remote timescale, and that this frequency
+/// drift is assumed to be the result from a (limit of) a random walk
+/// process (wiener process). Under this assumption, a timechange from t1
+/// to t2 has a state propagation matrix
+/// 1 (t2-t1)
+/// 0    0
+/// and a noise matrix given by
+/// v*(t2-t1)^3/3 v*(t2-t1)^2/2
+/// v*(t2-t1)^2/2   v*(t2-t1)
+/// where v is a constant describing how much the frequency drifts per
+/// unit of time.
+///
+/// This modules input consists of measurements containing:
+///  - the time of the measurement tm
+///  - the measured offset d
+///  - the measured transmission delay r
+/// On these, we assume that
+///  - there is no impact from frequency differences on r
+///  - individual measurements are independent
+///
+/// This information bare is not enough to feed the kalman filter. For
+/// this, two further pieces of information are needed: a measurement
```

Comment:
you go on to give only one of the pieces needed

Reply:
fixed